### PR TITLE
feat: Implement comprehensive postal code handling and validation

### PR DESCRIPTION
**Pull request by isaacchua0309**
This update ensures proper handling, validation, and formatting of postal codes throughout the application, addressing data integrity and preventing null reference errors. The changes include model updates, command enhancements, storage updates, and expanded test coverage.

## Key Changes:
- **Core Models**:
  - Updated Person and Location for postal code handling
  - Added validation and error messaging for postal codes in Person, Location, and LocationUtil

- **Command and Parser Updates**:
  - Enhanced  EditCommand, EditCommandParser, and AddressBookParser classes to handle postal code input, validation, and parsing

- **Data Handling**:
  - Updated JsonAdaptedPerson to handle postal codes in JSON serialization and validation
  - Modified test data to include postal codes

- **Test Coverage**:
  - Added tests for postal code validation, edge cases, and error handling across various test classes

- **Code Quality**:
  - Fixed checkstyle issues, including line length, trailing whitespace, and formatting inconsistencies

This update ensures that postal codes are consistently validated and handled across all relevant components of the application.

### CODE REVIEW

File: src/test/java/seedu/address/commons/core/VersionTest.java [hunk 1]
```diff
@@ -94,12 +94,12 @@ public void versionComparable_validVersion_compareToIsCorrect() {
...
         assertTrue(one.compareTo(another) < 0);
 
-        // Tests early access lower version vs not early access higher version compare by version number first
+        // Tests early access lower version vs not early access higher version compare by version number firs
         one = new Version(2, 15, 0, true);
         another = new Version(2, 15, 5, false);
         assertTrue(one.compareTo(another) < 0);
 
-        // Tests early access higher version vs not early access lower version compare by version number first
+        // Tests early access higher version vs not early access lower version compare by version number firs
```
Comment (noahang) [inline src/test/java/seedu/address/commons/core/VersionTest.java:102]: typo lol

File: src/test/java/seedu/address/commons/core/VersionTest.java [hunk 2]
```diff
@@ -94,12 +94,12 @@ public void versionComparable_validVersion_compareToIsCorrect() {
         another = new Version(2, 15, 0, false);
         assertTrue(one.compareTo(another) < 0);
 
-        // Tests early access lower version vs not early access higher version compare by version number first
+        // Tests early access lower version vs not early access higher version compare by version number firs
```
Comment (noahang) [inline src/test/java/seedu/address/commons/core/VersionTest.java:97]: typo lol

File: src/test/java/seedu/address/commons/util/CollectionUtilTest.java [hunk 3]
```diff
@@ -18,7 +18,7 @@ public void requireAllNonNullVarargs() {
         // no arguments
         assertNullPointerExceptionNotThrown();
 
-        // any non-empty argument list
+        // any non-empty argument lis
```
Comment (noahang) [inline src/test/java/seedu/address/commons/util/CollectionUtilTest.java:21]: lmao why all your "t"s disappearing

File: src/test/java/seedu/address/commons/util/JsonUtilTest.java [hunk 4]
```diff
@@ -39,7 +39,7 @@ public void deserializeObjectFromJsonFile_noExceptionThrown() throws IOException
         assertEquals(serializableTestClass.getMapOfIntegerToString(), SerializableTestClass.getHashMapTestValues());
     }
 
-    //TODO: @Test jsonUtil_readJsonStringToObjectInstance_correctObject()
+    //TODO: @Testt jsonUtil_readJsonStringToObjectInstance_correctObject()
```
Comment (noahang) [inline src/test/java/seedu/address/commons/util/JsonUtilTest.java:42]: LOL what is it with you and "t"s being in the wrong place

Comment (noahang) [reply src/test/java/seedu/address/commons/util/JsonUtilTest.java:42]: this is so funny I cannot

File: src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json
Comment (noahang) [inline src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json:1]: thank you for catching this for me too :D

File: src/test/java/seedu/address/commons/util/LocationUtilTest.java
Comment (noahang) [inline src/test/java/seedu/address/commons/util/LocationUtilTest.java:1]: @SomneelSaha2004 gonna need you to review this

File: src/test/java/seedu/address/commons/util/StringUtilTest.java [hunk 5]
```diff
@@ -126,7 +126,7 @@ public void containsWordIgnoreCase_validInputs_correctResult() {
     //---------------- Tests for getDetails --------------------------------------
 
     /*
-     * Equivalence Partitions: null, valid throwable object
+     * Equivalence Partitions: null, valid throwable objec
```
Comment (noahang) [inline src/test/java/seedu/address/commons/util/StringUtilTest.java:129]: typo

File: src/test/java/seedu/address/logic/commands/EditPersonDescriptorTest.java
Comment (noahang) [inline src/test/java/seedu/address/logic/commands/EditPersonDescriptorTest.java:1]: solid job

File: src/test/java/seedu/address/logic/parser/AddSportCommandParserTest.java
Comment (noahang) [inline src/test/java/seedu/address/logic/parser/AddSportCommandParserTest.java:1]: more typos

File: src/test/java/seedu/address/logic/parser/ArgumentTokenizerTest.java
Comment (noahang) [inline src/test/java/seedu/address/logic/parser/ArgumentTokenizerTest.java:1]: more typos

File: src/test/java/seedu/address/logic/parser/DeleteSportCommandParserTest.java
Comment (noahang) [inline src/test/java/seedu/address/logic/parser/DeleteSportCommandParserTest.java:1]: more typos

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 6]
```diff
@@ -28,7 +28,10 @@ public class ParserUtilTest {
     private static final String INVALID_PHONE = "+651234";
     private static final String INVALID_ADDRESS = " ";
 
-    private static final String INVALID_POSTAL_CODE = " ";
+    private static final String INVALID_POSTAL_CODE = "999999"; // Non-existent postal code
+    private static final String INVALID_POSTAL_CODE_FORMAT = "12AB34"; // Invalid forma
+    private static final String INVALID_POSTAL_CODE_TOO_SHORT = "12345"; // Too shor
```
Comment (noahang) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:33]: typo

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java
Comment (noahang) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:1]: YES WELL DONE THAT'S WHY YOU'RE HEAD OF TESTING

File: src/test/java/seedu/address/model/person/EmailTest.java
Comment (noahang) [inline src/test/java/seedu/address/model/person/EmailTest.java:1]: more typos

File: src/test/java/seedu/address/model/person/LocationTest.java
Comment (noahang) [inline src/test/java/seedu/address/model/person/LocationTest.java:1]: @SomneelSaha2004 need you to review this too

File: src/test/java/seedu/address/model/person/PersonTest.java
Comment (noahang) [inline src/test/java/seedu/address/model/person/PersonTest.java:1]: LGTM

File: src/test/java/seedu/address/storage/JsonUserPrefsStorageTest.java [hunk 7]
```diff
@@ -108,7 +108,7 @@ public void saveUserPrefs_allInOrder_success() throws DataLoadingException, IOEx
         Path pefsFilePath = testFolder.resolve("TempPrefs.json");
         JsonUserPrefsStorage jsonUserPrefsStorage = new JsonUserPrefsStorage(pefsFilePath);
 
-        //Try writing when the file doesn't exist
+        //Try writing when the file doesn't exis
```
Comment (noahang) [inline src/test/java/seedu/address/storage/JsonUserPrefsStorageTest.java:111]: typo

File: src/test/java/seedu/address/logic/commands/EditCommandTest.java [hunk 8]
```diff
@@ -136,7 +136,7 @@ public void execute_invalidPersonIndexUnfilteredList_failure() {
     public void execute_invalidPersonIndexFilteredList_failure() {
         showPersonAtIndex(model, INDEX_FIRST_PERSON);
         Index outOfBoundIndex = INDEX_SECOND_PERSON;
-        // ensures that outOfBoundIndex is still in bounds of address book list
+        // ensures that outOfBoundIndex is still in bounds of address book lis
```
Comment (noahang) [inline src/test/java/seedu/address/logic/commands/EditCommandTest.java:139]: typo

File: src/test/java/seedu/address/logic/commands/DeleteSportCommandTest.java [hunk 9]
```diff
@@ -57,7 +57,7 @@ public void execute_noSportFound_throwsCommandException() {
         Person personToModify = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
         Sport sportToDelete = personToModify.getSports().get(0);
 
-        // First delete existing sport
+        // First delete existing spor
```
Comment (noahang) [inline src/test/java/seedu/address/logic/commands/DeleteSportCommandTest.java:60]: typo

File: src/test/java/seedu/address/logic/commands/DeleteCommandTest.java [hunk 10]
```diff
@@ -71,7 +71,7 @@ public void execute_invalidIndexFilteredList_throwsCommandException() {
         showPersonAtIndex(model, INDEX_FIRST_PERSON);
 
         Index outOfBoundIndex = INDEX_SECOND_PERSON;
-        // ensures that outOfBoundIndex is still in bounds of address book list
+        // ensures that outOfBoundIndex is still in bounds of address book lis
```
Comment (noahang) [inline src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:74]: typo

File: src/test/java/seedu/address/logic/commands/AddSportCommandTest.java [hunk 11]
```diff
@@ -54,7 +54,7 @@ public void execute_duplicateSport_throwsCommandException() {
         Person personToModify = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
         Sport sportToAdd = new Sport("soccer");
 
-        // First add a sport
+        // First add a spor
```
Comment (noahang) [inline src/test/java/seedu/address/logic/commands/AddSportCommandTest.java:57]: typo

File: src/test/java/seedu/address/logic/LogicManagerTest.java
Comment (noahang) [inline src/test/java/seedu/address/logic/LogicManagerTest.java:1]: more typos

**Review by noahang — COMMENTED** [review]
I think just fix the typos, and maybe the requireAllNonNull(), overall solid job Isaac, thanks for cleaning up for me!

File: src/main/java/seedu/address/model/person/Person.java [hunk 12]
```diff
@@ -161,6 +164,7 @@ public String toString() {
                 .add("phone", phone)
                 .add("email", email)
                 .add("address", address)
+                .add("postalCode", getPostalCode())
                 .add("tags", tags)
```
[the remark this answers is not in the document]
Comment (SomneelSaha2004) [reply src/main/java/seedu/address/model/person/Person.java:174]: LGTM, modular
